### PR TITLE
Feature: Warn if destination directory is formatted as ntfs

### LINKED ISSUE #1854: Invalid argument (22) (NTFS target drive)
I was helping a co-worker setup BIT on his laptop. When he ran it the first time, he got a 'Invalid argument (22)' error.

Turns out his backup drive was formatted as NTFS.

I'm thinking it would be a good idea to add a check, on the destination directory, to ensure it's a compatible file system.

The question is, what constitutes a compatible file system?

I think the error 22 comes from trying to use certain characters in file names.

### PR DESCRIPTION
**Pull request by fallingrock**
Warn user if destination directory is formatted as ntfs to avoid "Unsupported operation" errors.

Fix #1854

### CODE REVIEW

File: qt/settingsdialog.py
Comment (buhtz) [inline qt/settingsdialog.py]: I wouldn't like to add a new dependency for that use case.

Why not using the shell tools?

    >>> r = subprocess.run(['df', '--output=fstype', '.'], stdout=subprocess.PIPE, text=True)
    >>> r.stdout.splitlines()[1]
    'ext4'

psutils also only read from the /proc filesystem.

**EDIT**:
BIT still hast this "feature". See https://github.com/bit-team/backintime/blob/8c7afef81ae79be491a848400261eabdc3e90c8c/common/tools.py#L1450

**Comment by fallingrock** [comment]
Hmmm ... build failed because psutil wasn't available.

Can I add psutil to the travis.yml file to fix this?

**Review by buhtz — CHANGES_REQUESTED** [review]
I wonder if this will help in all cases.

The file system depends on the way how the ntfs volume is mounted.
How was the volume mounted in case of the issue you reported? Would it catch that case?

File: CHANGES
Comment (buhtz) [inline CHANGES]:
```suggestion
* Feature: Warn if destination directory is formatted as NTFS (#1854) (David Gibbs @fallingrock)
```

File: qt/settingsdialog.py
Comment (fallingrock) [reply qt/settingsdialog.py]: Wasn't aware of the filesystem method ... I'll check it out.

File: qt/settingsdialog.py
Comment (buhtz) [inline qt/settingsdialog.py]: Imagine our translators using the web-frontend at translators.codeberg.org. Some of them have no coding background. They also need more context. Because of that it is important to give placeholders good names. The details are [hidden in our docs](https://github.com/bit-team/backintime/blob/dev/common/doc-dev/2_localization.md#instructions-for-the-translation-process).

```suggestion
                text = _('Warning: {path} is formatted as NTFS which has known incompatibilities '
                         'with Unix style file systems.'
                         'Are you sure you want to use it as your backup destination?')
                question = text.format(path=path)
```

I also removed the two `\n` because of our translators. I am not sure if, so please checkout if the dialog box does wrap the text by itself. And while you are checking it please create a screenshot of message box and attach it to the PR. If the text is not wrapped, please simply make two translatable strings like this:

    text = _('foo') + '\n\n' + _('bar')

Beside the text, the translators will also get a screenshot of the message box. So they will understand.

It is damn hard to get translators and keep them motivated. So I try to make it as easy as possible for them. 😄

**Review by buhtz — CHANGES_REQUESTED** [review]

**Comment by buhtz** [comment]
The message seems a bit unspecific for some users. Are there cases where NTFS volumes do work? So why are the users free to use NTFS with BIT? Shouldn't we just deny such destination volumes?

**Comment by fallingrock** [comment]
> The message seems a bit unspecific for some users. Are there cases where NTFS volumes do work? So why are the users free to use NTFS with BIT? Shouldn't we just deny such destination volumes?

I think it depends on if the user has filenames containing certain characters.

The following reserved characters:   
 \> (less than)
 \< (greater than)   
 : (colon)   
 " (double quote)   
 / (forward slash)   
 \ (backslash)   
 | (vertical bar or pipe)   
 ? (question mark)   
 \* (asterisk)

**Comment by fallingrock** [comment]
Screenshot of new error messages
image
image

**Comment by buhtz** [comment]
> I think it depends on if the user has filenames containing certain characters.
> The following reserved characters: ...

Interesting. In this case my alternative suggested approach just doing an rsync test run and check after that if hardlinks appear won't have caught such a situation. Your approach seems better.

Would you like to add an entry in FAQ.md about that NTFS details? We later can move that FAQ entry somewhere else into the docu.

File: common/config.py
Comment (buhtz) [inline common/config.py]: I am not skilled with English words. I just asked the "stochastic parrot" and modified the string a bit. So feel free to modify it either.

```suggestion
                _('The destination filesystem for {path} is formatted with NTFS, '
                  'which has known incompatibilities with Unix-style filesystems.')
```

File: qt/settingsdialog.py
Comment (buhtz) [inline qt/settingsdialog.py]: Beside avoiding escape characters like `\n` in the translation string I also try to avoid referencing user in person via "you".

```suggestion
                text = '\n'.join([
                       _('Warning: The destination filesystem for {path} is formatted with NTFS, '
                         'which has known incompatibilities with Unix-style filesystems.'),                             
                       _('Is this the backup destination to be used?')
                })
```

**Review by buhtz — CHANGES_REQUESTED** [review]
The PR looks good. Thank you very much for your efforts. A FAQ entry would be the cherry on top.

I would like to wait for review from the other team mates: emtiu and/or aryoda. They are not quit available currently so it can take some time.

File: common/config.py
Comment (fallingrock) [reply common/config.py]: I'm no english major either (GIBBeriSh is my family language :) ) ... I was just matching the existing message pattern in config.py.

**Review by buhtz — APPROVED** [review]

**Pull request merged**
